doc/lua_api.txt

```diff
@@ -1118,54 +1118,109 @@ A box of a regular node would look like:
 
     {-0.5, -0.5, -0.5, 0.5, 0.5, 0.5},
 
+Perlin noise
+------------
+Perlin noise creates a continuously-varying value depending on the input values.
+Usually in Minetest the input values are either 2D or 3D co-ordinates in nodes.
+The result is mostly used during mapgen to create the terrain shape, or to vary
+the density of decorations or vary the structure of ores.
+
+### Structure of perlin noise
+An 'octave' is a simple noise generator that outputs a value between -1 and 1.
+The smooth wavy noise it generates has a single characteristic scale, almost
+like a 'wavelength', so on its own does not create fine detail.
+So perlin noise combines several octaves to create variation on multiple scales.
+Each additional octave has a smaller 'wavelength' than the previous, often half
+the size of the previous (if `lacunarity` = 2.0, see below).
+
+This results in noise varying very roughly between -2.0 and 2.0, with an average
+value of 0, so `scale` and `offset` are then used to multiply and offset the
+noise variation. 
+
+The final perlin noise variation is created as follows:
+
+noise = offset + scale * (octave1 +
+                          octave2 * persistence +
+                          octave3 * persistence ^ 2 +
+                          octave4 * persistence ^ 3 +
+                          ...)
+
+(The word 'octave' is used because each additional simple noise generator often
```

Comment:
(Very long sentences in parentheses do not make much sense, please shorten it or remove the parentheses as it's still helpful information.)

Reply:
Ok.

Reply:
Not sure it's necessary, but it's interesting, not sure about keeping this.

Reply:
Very interesting, but also not sure. lua_api is already massive without adding in points of interest, but on the other hand this might help people understand. Either way, I agree about the brackets.

Reply:
Removed.